test/compress/destructuring.js

```diff
@@ -639,3 +639,25 @@ issue_2044_ecma_6_beautify: {
     }
     expect_exact: "({x: a = 1, y = 2 + b, z = 3 - c} = obj);"
 }
+
+issue_2140: {
+    options = {
+        unused: true,
+    }
+    input: {
+        !function() {
+            var s = {}, t = {};
+            [t.a] = [42];
+            console.log(t.a);
```

Comment:
That extra `console.log(t.a)` defeats the trap for `unused` to malfunction.

Fixed in https://github.com/mishoo/UglifyJS2/pull/2146/commits/c291f2d2aa03014401b97b07f420abd37f5f3ca2